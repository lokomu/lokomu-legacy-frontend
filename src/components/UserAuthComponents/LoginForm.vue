<template>
  <div
    class="rounded-3xl overflow-hidden max-w-md w-full bg-white dark:bg-gray-900"
  >
    <div class="mx-auto w-full xs:w-3/4 flex flex-col gap-4 p-4">
      <IHeader @click="$emit('closesModal')" />
      <!-- Email -->
      <div :class="{ error: v$.user.email.$errors.length }">
        <InputField
          id="loginEmail"
          v-model="v$.user.email.$model"
          type="email"
          name="email"
          :placeholder="t('placeholder.email')"
          :is-valid="v$.user.email.$dirty ? !v$.user.email.$error : null"
          @enter="loginClicked()"
        />
        <!-- error message for email -->
        <div
          v-for="(error, index) of v$.user.email.$errors"
          :key="index"
        >
          <div
            v-show="showError"
            id="emailErrorID"
            class="text-red-500 text-sm text-center"
          >
            {{ error.$message }}
          </div>
        </div>
      </div>
      <!-- Password -->
      <div :class="{ error: v$.user.password.$errors.length }">
        <InputField
          id="loginPassword"
          v-model="v$.user.password.$model"
          type="password"
          name="password"
          :placeholder="t('placeholder.password.default')"
          :is-valid="v$.user.password.$dirty ? !v$.user.password.$error : null"
          @enter="loginClicked()"
        />
        <!-- error message for password -->
        <div
          v-for="(error, index) of v$.user.password.$errors"
          :key="index"
          class="text-red-500 text-sm"
        >
          <div
            v-show="showError"
            id="passwordErrorID"
            class="text-red-500 text-sm text-center"
          >
            {{ error.$message }}
          </div>
        </div>
      </div>

      <!-- Button for logging in -->
      <ColorButton
        class="mx-auto"
        button-color="green"
        :text="t('button.login')"
        @click="loginClicked"
      />
      <div
        class="inline-block mx-auto text-blue-700 dark:text-blue-100 text-xs hover:font-bold cursor-pointer"
        @click="emits('openRegister')"
      >
        {{ t("button.signup_message") }}
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { reactive } from "vue";
import { ref } from "vue";
import useVuelidate from "@vuelidate/core";
import { required, email, helpers } from "@vuelidate/validators";
import { login } from "@/api/AuthService";
import { useUserStore } from "@/stores/UserStore";
import ColorButton from "@/components/BaseComponents/ColorButton.vue";
import { useI18n } from "vue-i18n";
import InputField from "@/components/BaseComponents/InputField.vue";
import IHeader from "@/components/NavigationComponents/IHeader.vue";
import { useNotification } from "@kyvg/vue3-notification";
import { getAdminList } from "@/api/factories/UserCommunityFactory";

const emits = defineEmits(["closesModal", "openRegister"]);

const userStore = useUserStore();
const { t } = useI18n();
const notifications = useNotification();

const user = reactive({
  email: "",
  password: "",
});
const showError = ref(false);

const rules = {
  user: {
    email: {
      required: helpers.withMessage(t("errors.login.email_empty"), required),
      email: helpers.withMessage(t("errors.login.email_invalid"), email),
    },
    password: {
      required: helpers.withMessage(t("errors.login.password"), required),
    },
  },
};

const v$ = useVuelidate(rules, { user });

async function loginClicked() {
  showError.value = true;
  v$.value.$touch();
  if (v$.value.$invalid) return;

  const loginRequest = {
    email: user.email.trim().toLowerCase(),
    password: user.password,
  };

  const loginResponse = await login(loginRequest);

  if (loginResponse.isLoggedIn === true) {
    userStore.login(loginResponse.authResponse);
    const adminList = await getAdminList();
    userStore.addAdminList(adminList);
  } else {
    notifications.notify({
      type: "error",
      text: t("errors.login.wrong"),
    });
  }
}
</script>
