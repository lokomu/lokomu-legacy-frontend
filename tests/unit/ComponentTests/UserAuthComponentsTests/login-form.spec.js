import { mount } from "@vue/test-utils";
import LoginForm from "@/components/UserAuthComponents/LoginForm.vue";
import { describe, expect, it, beforeEach } from "vitest";

describe("LoginForm component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(LoginForm);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
