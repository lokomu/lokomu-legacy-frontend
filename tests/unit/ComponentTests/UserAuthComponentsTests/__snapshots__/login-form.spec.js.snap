// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginForm component > renders correctly 1`] = `
<div
  class="rounded-3xl overflow-hidden max-w-md w-full bg-white dark:bg-gray-900"
>
  <div
    class="mx-auto w-full xs:w-3/4 flex flex-col gap-4 p-4"
  >
    <div
      class="py-2 flex justify-center"
    >
      <a
        class="flex cursor-pointer"
      >
        <svg
          class="fill-green-500 dark:fill-white h-10 mr-2"
          viewBox="0 0 40 18.4"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M38.2 18.4c-2.5 0-4.5-2-4.5-4.5V0H36v13.9c0 1.2 1 2.1 2.1 2.1v2.4zM30.6 16.2a7.6 7.6 0 0 1-10.7-.2l-.2-.2c-3.7-4-1.5-11 3.8-12.2 6.4-1.8 10.2 6 4.7 9.5-1.7 1-3.9.7-5.2-.4 5.2 2.2 8.4-5 3.1-6.9-4.6-.9-7.7 5-4.6 8.4 1.8 2.2 5.5 2.3 7.5.3l1.6 1.7zM15.6 0v11.1c-.3 7.7-10.5 10-14.1 3.3C-1.5 9.3 3.4 2.8 9 4.3c1 .2 1.9.7 2.6 1.3a6 6 0 0 0-5.1-.1c-4.7 1.9-4.5 9.3.6 10.4 3 .7 6.2-1.8 6.2-4.9V0h2.3z"
          />
        </svg>
        <svg
          class="mt-1 h-9 fill-green-500 dark:fill-white"
          viewBox="0 0 40 20.6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m28.4 12.1 1.7.8c.7.3 1.4.5 2.1.5 2.9.2 5.6-2.1 5.6-5.1 0-2.2-1.5-4.2-3.7-4.7a5 5 0 0 0-4 1.2l-.7.7-11.7 11.9c-.3.3-.8.8-1.2 1A10.2 10.2 0 0 1 0 11.1C-.3 6.1 3.3 1.4 8.3.9c2.6-.3 5 .5 7 2.1l3.3 2.6-3.8-1.9A7.8 7.8 0 0 0 4.6 5.3a7 7 0 0 0 1.3 10 6.5 6.5 0 0 0 7.1.3 438.6 438.6 0 0 1 13.8-13l.8-.7a7.5 7.5 0 0 1 12 2.8 7 7 0 0 1-3.1 8.5c-2 1-4.7 1-6.5-.3"
          />
        </svg>
      </a>
    </div>
    <!-- Email -->
    <div
      class=""
    >
      <input
        class="w-full text-center text-sm p-2.5 outline-none rounded-lg bg-gray-100 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500 focus:border-green-500 focus:ring-2 focus:ring-green-500 focus:ring-opacity-50 dark:focus:border-green-500"
        id="loginEmail"
        name="email"
        placeholder="Email"
        type="email"
      />
      <!-- error message for email -->
      
      
    </div>
    <!-- Password -->
    <div
      class=""
    >
      <input
        class="w-full text-center text-sm p-2.5 outline-none rounded-lg bg-gray-100 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500 focus:border-green-500 focus:ring-2 focus:ring-green-500 focus:ring-opacity-50 dark:focus:border-green-500"
        id="loginPassword"
        name="password"
        placeholder="Password"
        type="password"
      />
      <!-- error message for password -->
      
      
    </div>
    <!-- Button for logging in -->
    <button
      class="flex bg-gradient-to-br hover:bg-gradient-to-tl items-center text-white font-medium rounded-lg text-sm px-5 py-2.5 focus:ring-4 focus:outline-none text-center from-green-400 via-green-600 to-green-700 focus:ring-green-300 dark:focus:ring-green-800 mx-auto"
      type="button"
    >
      Log In
    </button>
    <div
      class="inline-block mx-auto text-blue-700 dark:text-blue-100 text-xs hover:font-bold cursor-pointer"
    >
      New user? Sign up
    </div>
  </div>
</div>
`;
